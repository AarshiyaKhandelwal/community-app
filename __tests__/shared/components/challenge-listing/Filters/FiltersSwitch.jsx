import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import FiltersSwitch from 'components/challenge-listing/Filters/FiltersSwitch';

const onSwitch = jest.fn();

const mockDatas = [{
  active: false,
  filtersCount: 0,
  onSwitch,
  className: 'name',
}, {
  active: true,
  filtersCount: 5,
  onSwitch,
  className: 'name',
}];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <FiltersSwitch {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <FiltersSwitch {...this.props} />;
  }
}

test('handle events', () => {
  const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[1]} />));
  const matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('Switch') && item.className.match('tc-outline-btn'));
  expect(matches).toHaveLength(1);

  TU.Simulate.click(matches[0]);
  expect(onSwitch).toHaveBeenCalledTimes(1);
});

