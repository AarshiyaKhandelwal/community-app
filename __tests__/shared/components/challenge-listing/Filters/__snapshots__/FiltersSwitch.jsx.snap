// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="tc-outline-btn name src-shared-components-challenge-listing-Filters-FiltersSwitch-___style__FiltersSwitch___2ixiH"
  onClick={[Function]}
>
  <FiltersIcon
    className="src-shared-components-challenge-listing-Filters-FiltersSwitch-___style__FiltersIcon___OqBJh"
    color="#5D5D66"
  />
  Filters
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="tc-outline-btn name src-shared-components-challenge-listing-Filters-FiltersSwitch-___style__FiltersSwitch___2ixiH src-shared-components-challenge-listing-Filters-FiltersSwitch-___style__active___2wLTl"
  onClick={[Function]}
>
  <FiltersIcon
    className="src-shared-components-challenge-listing-Filters-FiltersSwitch-___style__FiltersIcon___OqBJh"
    color="#5D5D66"
  />
  Filters
  <span
    className="src-shared-components-challenge-listing-Filters-FiltersSwitch-___style__filtersCount___2MZij"
  >
    5
  </span>
</div>
`;
